src/Workouts.svelte

```diff
@@ -14,30 +14,48 @@
         let baseWeight = Math.min(...c.weights);
 
         for (let weight of c.weights) {
-            let reps = c.proportionateReps ? Math.floor(baseWeight / weight * c.baseReps) : c.baseReps;
-
+            let base_reps = c.proportionateReps ? Math.floor(baseWeight / weight * c.baseReps) : c.baseReps;
+            let max_volume = 0;
             for (let sets = c.minSets; sets <= c.maxSets; sets += c.delta) {
+                // volume cycle: Reps = base reps
+                const reps = base_reps;
+                const volume = weight * sets * reps;
                 workouts.push({
                     weight,
                     sets,
                     reps,
-                    volume: weight * sets * reps
+                    volume,
+                    is_volume_cycle: true
                 });
+                max_volume = Math.max(max_volume, volume);
             }
 
             if (c.includeDensity) {
-                for (let sets = c.maxSets - c.delta, i = 1; sets >= c.minSets; sets -= c.delta, i++) {
+                for (let sets = c.maxSets - c.delta; sets >= c.minSets; sets -= c.delta) {
+                    const reps = Math.floor(max_volume/(sets * weight));
```

Comment:
i think Math.round is more appropriate here - it's ok to be slightly above the max volume.